controllers/capabilities/proxyconfigpromote_controller.go

```diff
@@ -162,15 +181,13 @@ func (r *ProxyConfigPromoteReconciler) proxyConfigPromoteReconciler(proxyConfigP
 			}
 			latestStagingVersion = stageElement.ProxyConfig.Version
 
-			// Fetch production state, if product has not been promoted to production yet the state would be 0.
-			productionElement, err := threescaleAPIClient.GetLatestProxyConfig(productIDStr, "production")
```

Comment:
I think this is still required.
Reason why is the following scenario:
- I have a configuration in stage (let's say version 4) and production (at version 3).
- I create a proxy config promote with stage bump to version 5.
- During reconciliation, if I pass the stage check and it turns out that both of my versions are different, the stage bump is successfull
- We are then moving to line 234 and sending in the latestStage and latestProduction version, in your approach, the production is always at 0 - ignored. 
- The reason why we sometimes might want to keep the Pcp CR is for tracking, and if so, a production version should also be set therefore we should not be sending nil value if it's not nil.

Reply:
Done